test/governance/candidacy.users.test.ts

```diff
@@ -0,0 +1,233 @@
+import { getEnvironmentRequiredVars, waitForNBlocks } from "../../utils/utils";
+import { testLog } from "../../utils/Logger";
+
+// Global variables
+const { chainUri: environmentUri } = getEnvironmentRequiredVars();
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.setTimeout(15000000);
+
+// Chain variables
+let bootnode: Node;
+let keyring: Keyring;
+let sudo: SudoUser;
+let termDuration: number; // Set as a const in mangata-node
+const termBlockDelta = 3; // A small block delta at term boundaries
+
+// Test variables
+let candidate: GovernanceUser;
+let voter: GovernanceUser;
+let bank: Bank;
+const stakeAmount = new BN(Math.pow(10, 16).toString());
+const accountFundingAmount = new BN(Math.pow(10, 17).toString());
+
+// Cross-test state
+const state: {
+  accounts: [GovernanceUser | null];
+} = { accounts: [null] };
+
+beforeAll(async () => {
+  await cryptoWaitReady(); // Wait for Polkadots WASM backend
+
+  // Instantiate nodes
+  bootnode = new Node(environmentUri);
+  await bootnode.connect();
+  await bootnode.subscribeToHead();
+
+  // Set chain variables
+  termDuration = (
+    await bootnode.api!.derive.elections.info()
+  ).termDuration.toNumber();
+  keyring = new Keyring({ type: "sr25519" });
+  sudo = UserFactory.createUser(Users.SudoUser, keyring, bootnode) as SudoUser;
+});
+
+beforeEach(async () => {
+  // Don't run tests at the end of a term
+  if (termDuration % bootnode.lastBlock!) {
+    waitForNextTerm();
+  }
+
+  // Initialize actors
+  candidate = UserFactory.createUser(
+    Users.GovernanceUser,
+    keyring,
+    bootnode
+  ) as GovernanceUser;
+  voter = UserFactory.createUser(
+    Users.GovernanceUser,
+    keyring,
+    bootnode
+  ) as GovernanceUser;
+  bank = new Bank(sudo);
+
+  // Record new test state
+  state.accounts.push(voter);
+
+  // Fund accounts
+  await voter.addMGATokens(bank.sudoUser, accountFundingAmount);
+  await candidate.addMGATokens(bank.sudoUser, stakeAmount);
+
+  // Subscribe to events
+  candidate.node.subscribeToUserBalanceChanges(candidate);
+  voter.node.subscribeToUserBalanceChanges(voter);
+});
+
+afterAll(async () => {
+  // Tests in files are unlikely to impact each other
+  // but not clearing up over multiple test files will break
+  await cleanState();
+});
+
+describe("Governance -> Candidacy -> Users", () => {
+  test("User can run for candidacy", async () => {
+    const candidatesAddress = candidate.keyRingPair.address;
+
+    await candidate.runForCouncil();
+    await waitForNBlocks(2);
+
+    const termZero = getLastBlocksElectionData();
+
+    expect(flattenArray(termZero.Candidates)).toEqual(
+      expect.arrayContaining([candidatesAddress])
+    );
+  });
+
+  test("Candidate can renounce candidacy", async () => {
+    const candidatesAddress = candidate.keyRingPair.address;
+
+    await candidate.runForCouncil();
+    await waitForNBlocks(2);
+
+    await candidate.renounceCandidacy({
+      Candidate: candidate.keyRingPair.address,
```

Comment:
@goncer All tests are working except an extrinsic error on this function call!

```
Summary of all failing tests
 FAIL  test/governance/candidacy.users.test.ts (1460.643 s)
  ● Governance -> Candidacy -> Users › Candidate can renounce candidacy

    createType(Call):: Call: failed decoding elections.renounceCandidacy:: Struct: failed on args: {"renouncing":"Renouncing"}:: Struct: failed on renouncing: {"_enum":{"Member":"Null","RunnerUp":"Null","Candidate":"Compact<u32>"}}:: Enum(Candidate):: u32: Input too large. Found input with 159 bits, expected 32

      29 |   async renounceCandidacy(candidateStatus: any): Promise<void> {
      30 |     await signSendAndWaitToFinishTx(
    > 31 |       this.api.tx.elections.renounceCandidacy(candidateStatus),
         |                             ^
      32 |       this.keyRingPair
      33 |     );
      34 |   }

      at createTypeUnsafe (node_modules/@polkadot/types/create/createType.js:71:11)
      at createType (node_modules/@polkadot/types/create/createType.js:83:10)
      at TypeRegistry.createType (node_modules/@polkadot/types/create/registry.js:257:39)
      at extrinsicFn (node_modules/@polkadot/metadata/decorate/extrinsics/createUnchecked.js:20:21)
      at Object.renounceCandidacy (node_modules/@polkadot/api/base/Decorate.js:324:46)
      at GovernanceUser.renounceCandidacy (utils/Framework/User/GovernanceUser.ts:31:29)
      at Object.<anonymous> (test/governance/candidacy.users.test.ts:113:21)


Test Suites: 1 failed, 1 of 48 total
Tests:       1 failed, 3 passed, 4 total
Snapshots:   0 total
Time:        1461.277 s
```

Reply:
Plz, read the error description. 
```
 createType(Call):: Call: failed decoding elections.renounceCandidacy:: Struct: failed on args: {"renouncing":"Renouncing"}:: Struct: failed on renouncing: {"_enum":{"Member":"Null","RunnerUp":"Null","Candidate":"Compact<u32>"}}:: Enum(Candidate):: u32: Input too large. Found input with 159 bits, expected 32
```
This is telling you that something is wrong with the parameter, not an extrinsic result. 
I think that the issue is that you are adding the address as CandidacyParamter and as you can see in the error, the Candidate object accepts only a number. ( compact<u32> }
https://docs.google.com/document/d/1z26Cy9SLvCZEPt7LfIfgcseBWvsUaHUupE34GBs7wps/edit?disco=AAAASXHODns
Most prob a copy-paste issue. :)